src/main/java/org/assertj/core/api/AbstractByteBufferAssert.java

```diff
@@ -0,0 +1,498 @@
+   *
+   * @return {@code this} assertion object.
+   * @throws AssertionError if the actual {@code ByteBuffer} is {@code null}.
+   * @throws AssertionError if the actual {@code ByteBuffer} has not been flipped.
+   * @throws AssertionError if the actual {@code ByteBuffer}s content does not end with the given expected string.
+   */
+  public SELF endsWith(String expected, Charset charset) {
+    isFlipped();
+
+    String contentString = new String(getContent(actual), charset);
+    if (!contentString.endsWith(expected)) throwAssertionError(contentsShouldEndWith(expected, actual, charset));
+    return myself;
+  }
+
+  /**
+   * Verifies that the content of the actual {@code ByteBuffer} ends with the given expected byte array.
+   *
+   * Example:
+   *
+   * <pre><code class='java'> ByteBuffer buffer = ByteBuffer.wrap("test".getBytes());
+   *
+   * // this assertion succeeds ...
+   * assertThat(buffer).endsWith("test".getBytes());
+   *
+   * // ... but this one fails as the content of "buffer" does not end with the given expected byte array.
+   * assertThat(buffer).endsWith("test two".getBytes());</code></pre>
+   *
+   * @return {@code this} assertion object.
+   * @throws AssertionError if the actual {@code ByteBuffer} is {@code null}.
+   * @throws AssertionError if the actual {@code ByteBuffer} has not been flipped.
+   * @throws AssertionError if the actual {@code ByteBuffer}s content does not end with the given expected byte array.
+   */
+  public SELF endsWith(byte[] expected) {
+    isFlipped();
+
+    byteArrays.assertEndsWith(info, getContent(actual), expected);
+    return myself;
+  }
+
+  /**
+   * Verifies that the content of the actual {@code ByteBuffer} ends with the given expected {@code ByteBuffer}.
+   *
+   * Example:
+   *
+   * <pre><code class='java'> ByteBuffer actual = ByteBuffer.wrap("test".getBytes());
+   *
+   * // this assertion succeeds ...
+   * ByteBuffer expected = ByteBuffer.wrap("test".getBytes());
+   * assertThat(actual).endsWith(expected);
+   *
+   * // ... but this one fails as the content of "buffer" does not end with the given expected {@code ByteBuffer}.
+   * ByteBuffer expected = ByteBuffer.wrap("test two".getBytes());
+   * assertThat(actual).endsWith(expected);</code></pre>
+   *
+   * @return {@code this} assertion object.
+   * @throws AssertionError if the actual {@code ByteBuffer} is {@code null}.
+   * @throws AssertionError if the actual {@code ByteBuffer} has not been flipped.
+   * @throws AssertionError if the actual {@code ByteBuffer}s content does not end with the given expected {@code ByeBuffer}.
+   */
+  public SELF endsWith(ByteBuffer expected) {
+    isFlipped();
+
+    byteArrays.assertEndsWith(info, getContent(actual), getContent(expected));
+    return myself;
+  }
+
+  private byte[] getContent(ByteBuffer buffer) {
+    byte[] content = new byte[buffer.limit()];
+    for (int i = 0; i < content.length; i++) {
+      content[i] = buffer.get();
```

Comment:
This operation starts consuming the buffer at the current position and rewinds its position to zero afterward. There's nothing wrong with the current position, however the buffer position should be reset to the same position.

something like:

```
int position = buffer.position();

// get cotent
buffer.position(position);
```

Also, retrieval could be slightly simplified to `buffer.get(content)` instead of looping.

Reply:
Good suggestion! It did made me question the following however:
The current implementation asserts from the current position i.e.:
```
ByteBuffer buffer = ByteBuffer.wrap("someString".getBytes())
buffer.startsWith("some") // this assertion will succeed
```
But that also means that moving the position of the buffer will change the outcome of the assertions, i.e.:
```
ByteBuffer buffer = ByteBuffer.wrap("someString".getBytes())
buffer.position(1)
buffer.startsWith("some") // this assertion will fail
```

I thought that this behaviour would perhaps be desirable, as it allows you to use the assertion methods as you move through the buffer, allowing for combinations like:
```
ByteBuffer buffer = ByteBuffer.wrap("header:someHeader;body:someBody".getBytes())
buffer.startsWith("header")
... perform header assertions
buffer.startsWith("body")
... perform body assertions
```

But should this indeed be the desired behaviour? Or would it be more useful to make all (or perhaps some?) of the assertions check the entire buffer instead? A third option would be to have assertion methods for both ways.

Reply:
Assertions should not alter the position IMO but rather be good citizens and remember the last position.

The second example should be the following code to make it work:

```
ByteBuffer buffer = ByteBuffer.wrap("header:someHeader;body:someBody".getBytes())
buffer.startsWith("header")
// …
buffer.position(18);
buffer.startsWith("body")
... perform body assertions
```

I suggest to come up with a minimal API for now and learn from feedback.

> But should this indeed be the desired behaviour

It makes sense from how one works with buffers, so yes, I think so.

> assertions check the entire buffer instead

These approaches do not contradict. Initially, the buffer is at position `0` and that allows running assertions against the entire buffer. An extension to the API could look like `assertThat(buffer).atPosition(0).startsWith(…)`. For now, I'd keep it as simple as possible.